hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieRealtimeTableCompactor.java

```diff
@@ -92,7 +92,8 @@ public HoodieCompactionMetadata compact(JavaSparkContext jsc, HoodieWriteConfig
     log.info("Total of " + operations.size() + " compactions are retrieved");
 
     // Filter the compactions with the passed in filter. This lets us choose most effective compactions only
-    operations = config.getCompactionStrategy().orderAndFilter(config, operations);
+    operations = config.getCompactionStrategy().orderAndFilter(config, operations.stream().filter(c ->
```

Comment:
Do the filter above when we collect the compaction operations.

Reply:
addressed.